itests/manual_cc_test.go

```diff
@@ -0,0 +1,216 @@
+package itests
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/builtin"
+	miner14 "github.com/filecoin-project/go-state-types/builtin/v14/miner"
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/go-state-types/proof"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/build"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/miner"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/itests/kit"
+)
+
+// Manually onboard CC sectors, bypassing lotus-miner onboarding pathways
+
+func TestManualCCOnboarding(t *testing.T) {
+	req := require.New(t)
+	// kit.Expensive(t)
+
+	kit.QuietMiningLogs()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var (
+		//provingPeriod abi.ChainEpoch = 2880
+		blocktime = 2 * time.Millisecond
+
+		client kit.TestFullNode
+		minerA kit.TestMiner // A has some genesis sector, just keeps power
+		minerB kit.TestMiner // B pledges some sector
+	)
+
+	opts := []kit.NodeOpt{kit.WithAllSubsystems()}
+	ens := kit.NewEnsemble(t, kit.MockProofs()).
+		FullNode(&client, opts...).
+		Miner(&minerA, &client, opts...).
```

Comment:
Right so we just give A some genesis sectors with power here but then don't add more sectors to A, right ?

Reply:
Yes, and there's an additional action item here at some stage to try and do this outside of the miner framework - there's both presealing and regular chain iteration happening behind the scenes for `minerA`. Not so important now, and maybe it just means tinkering with the test kit to decouple it a little more.